pywbem_mock/_defaultinstanceprovider.py

```diff
@@ -0,0 +1,527 @@
+        # if provider:
+        # return provider.DeleteInstance(
+        #    InstanceName)
+        #
+
+        return self.default_instance_provider.DeleteInstance(
+            InstanceName,)
+
+
+class DefaultInstanceProvider(BaseProvider):
+    """
+    This class defined those instance provider methods that may have user-
+    defined providers that override the default provider implementation in this
+    class.
+
+    User providers are defined by creating a subclass of this class and
+    defining a new provider method for one of the methods in this class with
+    the same signature.
+
+    Note that user-defined providers may, in turn, call the default providers
+    in this class.
+
+    """
+    def __init__(self):
+        """
+        """
+        pass
+
+    ####################################################################
+    #
+    #   CreateInstance server request
+    #
+    ####################################################################
+
+    def CreateInstance(self, namespace, NewInstance):
+        """
+        Implements a WBEM server responder for
+        :meth:`pywbem.WBEMConnection.CreateInstance`.
+
+        Create a CIM instance in the local CIM repository of this class.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            The name of the CIM namespace in the CIM repository (case
+            insensitive). Must not be `None`. Leading or trailing slash
+            characters are ignored.
+
+          NewInstance (:class:`~pywbem.CIMInstance`):
+            A representation of the CIM instance to be created.
+
+            The `classname` attribute of this object specifies the creation
+            class for the new instance.
+
+            The namespace of the `path` attribute must be `None` or the
+            same as the namespace parameter.
+
+            The `properties` attribute of this object specifies initial
+            property values for the new CIM instance.
+
+            Instance-level qualifiers have been deprecated in CIM, so any
+            qualifier values specified using the `qualifiers` attribute
+            of this object are ignored.
+
+        Returns:
+            CIMInstanceName with the server defined instance path
+            for the new instance.
+
+        Raises:
+
+            CIMError: CIM_ERR_ALREADY_EXISTS
+              The instance defined by namespace and instance name already
+              exists.
+
+            CIMError: CIM_ERR_INVALID_CLASS
+              The class defined in NewInstance  does not exist in the
+              namespace.
+        """
+
+        new_instance = NewInstance
+        if not isinstance(new_instance, CIMInstance):
+            raise CIMError(
+                CIM_ERR_INVALID_PARAMETER,
+                _format("NewInstance parameter is not a valid CIMInstance. "
+                        "Rcvd type={0}", type(new_instance)))
+
+        self.validate_namespace(namespace)
```

Comment:
This check could be done before calling the provider.

Reply:
Already commented above